src/cc/usdt/usdt.cc

```diff
@@ -273,6 +279,10 @@ std::string Context::resolve_bin_path(const std::string &bin_path) {
     ::free(which_so);
   }
 
+  if (!result.empty() && pid_) {
+    result = tfm::format("/proc/%d/root%s", *pid_, result);
```

Comment:
is this true all the time? Or it is just another try.

Reply:
`Context` can be initialized without a `pid`, and separately from that both `bcc_procutils_which` and `bcc_procutils_which_so` can fail due to diabolical input. Functions that call `Context::resolve_bin_path` rely on its return string to be empty as a signal of "I couldn't find a bin path" so we must only prepend `/proc/PID/root` when there's something to prepend to.